cupy/cuda/memory.pyx

```diff
@@ -675,6 +675,21 @@ def _set_thread_local_allocator(allocator):
     _thread_local.allocator = allocator
 
 
+cdef inline intptr_t _get_stream_identifier(intptr_t stream_ptr):
+    # When PTDS is enabled, return an ID to uniquely identify the default
+    # stream for each thread. (#5069)
+    if stream_ptr != runtime.streamPerThread:
+        return stream_ptr
+
+    cpdef intptr_t tid
+    try:
+        tid = _thread_local._tid
+    except AttributeError:
+        _thread_local._tid_obj = tid_obj = object()
+        _thread_local._tid = tid = id(tid_obj)
+    return -tid
```

Comment:
why is the reason because this is negative?
to differentiate pointers from ids?

Reply:
Yeah, to avoid conflicts with stream pointers I think.

Reply:
Yes.